pkg/provider/docker/config.go

```diff
@@ -345,7 +345,17 @@ func (p Provider) getIPAddress(ctx context.Context, container dockerData) string
 			logger.Warnf("Unable to get IP address for container %s : Failed to inspect container ID %s, error: %s", container.Name, connectedContainer, err)
 			return ""
 		}
-		return p.getIPAddress(ctx, parseContainer(containerInspected))
+
+		// Check connected container for traefik.docker.network, falling back to
+		// the network specified on the current container.
+		containerParsed := parseContainer(containerInspected)
+		extraConf, _ := p.getConfiguration(containerParsed)
```

Comment:
I'm not sure yet that you/we have got the right idea here (still thinking), but in any case, I think that line can instead just be:
```
extraConf := configuration{
	Docker: specificConfiguration{
		Network: p.Network,
	},
}
```
right?

Reply:
nevermind that comment too, we indeed need the full method in case a label overrides the value from p.Network.